plugins/src/android/OptimoveSDKPlugin.java

```diff
@@ -1,16 +1,205 @@
+                return true;
+            case GET_CURRENT_USER_IDENTIFIER:
+                this.getCurrentUserIdentifier(callbackContext);
+                return true;
+        }
         return false;
     }
+
+    private void setUserId(JSONArray args, CallbackContext callbackContext) {
+
+        try {
+            String userId = args.getString(0);
+            Optimove.getInstance().setUserId(userId);
+        } catch (Exception e) {
+            callbackContext.error(e.getMessage());
+            e.printStackTrace();
+            return;
+        }
+
+        callbackContext.success();
+
+    }
+
+    private void setUserEmail(JSONArray args, CallbackContext callbackContext) {
+        try {
+            String email = args.getString(0);
+            Optimove.getInstance().setUserEmail(email);
+        } catch (Exception e) {
+            callbackContext.error(e.getMessage());
+            e.printStackTrace();
+            return;
+        }
+        callbackContext.success();
+    }
+
+    private void reportEvent(JSONArray args, CallbackContext callbackContext) {
+        try {
+            reportEvent(args, callbackContext);
+            String eventName = args.getString(0);
+            JSONObject params = args.optJSONObject(1);
+            if (params == null) {
+                this.reportEvent(eventName, callbackContext);
+            } else {
+                reportEvent(eventName, params, callbackContext);
+            }
+        } catch (Exception e) {
+            callbackContext.error(e.getMessage());
+            e.printStackTrace();
+            return;
+        }
+
+    }
+
+    private void reportEvent(String eventName, CallbackContext callbackContext) {
+        try {
+            Optimove.getInstance().reportEvent(eventName);
+        } catch (Exception e) {
+            callbackContext.error(e.getMessage());
+            e.printStackTrace();
+            return;
+        }
+        callbackContext.success();
+
+    }
+
+    private void reportEvent(String eventName, JSONObject parameters, CallbackContext callbackContext) {
+        Map<String, Object> parametersMap;
+        try {
+            parametersMap = JsonUtils.toMap(parameters);
+        } catch (JSONException e) {
+            callbackContext.error(e.getMessage());
+            e.printStackTrace();
+            return;
+        }
+        try {
+            Optimove.getInstance().reportEvent(eventName, parametersMap);
+        } catch (Exception e) {
+            callbackContext.error(e.getMessage());
+            e.printStackTrace();
+            return;
+        }
+        callbackContext.success();
+
+    }
+
+    private void reportScreenVisit(JSONArray args, CallbackContext callbackContext) {
+        try {
+            String screenName = args.getString(0);
+            String screenCategory = args.optString(1);
+            if (screenCategory.equals("null")) {
+                reportScreenVisit(screenName, callbackContext);
```

Comment:
Do we need to have these other helpers to delegate to, seems like we can replace with Optimove.report... calls directly in here to simplify?